compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsPureTypeUniqueVisitor.java

```diff
@@ -68,6 +68,9 @@ private boolean isAnydata(BType type) {
             case TypeTags.JSON:
             case TypeTags.XML:
             case TypeTags.XML_TEXT:
+            case TypeTags.XML_ELEMENT:
+            case TypeTags.XML_COMMENT:
+            case TypeTags.XML_PI:
```

Comment:
Why was this change needed? Shouldn't the same be done in `IsAnydataUniqueVisitor`? 

`IsPureTypeUniqueVisitor` and pure types (what used to be `anydata|error`) are no longer applicable and we should be able to get rid of this. Shall we at least get rid of the usage of this in `IsAnydataUniqueVisitor` for now?

Reply:
Is there a test covering whatever this change was required for?

Reply:
This change was done because of a test failure. I did not know that we are trying to get rid of `IsPureTypeUniqueVisitor`. I also noticed that there are not many usages of `IsPureTypeUniqueVisitor`. Shall I send a separate PR to remove `IsPureTypeUniqueVisitor`?

Reply:
Shall we remove it from `IsAnydataUniqueVisitor` in this PR itself, because fixing the original issue required changes in `IsPureTypeUniqueVisitor`? Just replacing them with recursive calls to `IsAnydataUniqueVisitor` itself should work, right?

Reply:
Changed